core/src/main/scala/cats/data/Xor.scala

```diff
@@ -0,0 +1,374 @@
+package cats
+package data
+
+import scala.annotation.tailrec
+import scala.reflect.ClassTag
+import scala.util.{Failure, Success, Try}
+
+/** Represents a right-biased disjunction that is either an `A` or a `B`.
+ *
+ * An instance of `A [[Xor]] B` is either a `[[Xor.Left Left]][A]` or a `[[Xor.Right Right]][B]`.
+ *
+ * A common use of [[Xor]] is to explicitly represent the possibility of failure in a result as opposed to
+ * throwing an exception.  By convention, [[Xor.Left Left]] is used for errors and [[Xor.Right Right]] is reserved for successes.
+ * For example, a function that attempts to parse an integer from a string may have a return type of
+ * `NumberFormatException [[Xor]] Int`. However, since there is no need to actually throw an exception, the type (`A`)
+ * chosen for the "left" could be any type representing an error and has no need to actually extend `Exception`.
+ *
+ * `A [[Xor]] B` is isomorphic to `scala.Either[A, B]`, but [[Xor]] is right-biased, so methods such as `map` and
+ * `flatMap` apply only in the context of the "right" case. This right bias makes [[Xor]] more convenient to use
+ * than `scala.Either` in a monadic context. Methods such as `swap`, and `leftMap` provide functionality
+ * that `scala.Either` exposes through left projections.
+ *
+ * Some additional [[Xor]] methods can be found in [[Xor.XorOps XorOps]]. These methods are not defined on [[Xor]] itself because
+ * [[Xor]] is covariant in its types `A` and `B`.
+ */
+sealed abstract class Xor[+A, +B] extends Product with Serializable {
```

Comment:
I would have liked to add a `@deprecated` annotation here but it causes methods associated with type class instances (down below) to error since they are technically outside of the type.. :/